### PR TITLE
Add support for individual multicast address

### PR DESCRIPTION
**Pull request by graben**
@rhuss : PR for #421

### CODE REVIEW

File: agent/core/src/main/java/org/jolokia/config/ConfigKey.java [hunk 1]
```diff
@@ -197,6 +197,16 @@
      */
     DISCOVERY_AGENT_URL("discoveryAgentUrl",true,false),
 
+    /**
+     * IPv4 Address for Jolokia's Multicast group.
+     */
+    MULTICAST_GROUP("multicastGroup",true,false),
```
Comment (rhuss) [inline agent/core/src/main/java/org/jolokia/config/ConfigKey.java]:
```suggestion
    MULTICAST_GROUP("multicastGroup",true,false, "239.192.48.84"),
```

You can directly add a default value to the overloaded enum constructor with 4 args.

File: agent/core/src/main/java/org/jolokia/config/ConfigKey.java [hunk 2]
```diff
@@ -197,6 +197,16 @@
...
+
+    /**
+     * Multicast port where to listen for queries.
+     */
+    MULTICAST_PORT("multicastPort",true,false),
```
Comment (rhuss) [inline agent/core/src/main/java/org/jolokia/config/ConfigKey.java]:
```suggestion
    MULTICAST_PORT("multicastPort",true,false,"24884"),
```

File: agent/core/src/main/java/org/jolokia/discovery/MulticastUtil.java [hunk 3]
```diff
@@ -17,22 +18,40 @@
...
-    public static final int JOLOKIA_MULTICAST_PORT = 24884;
+    // Default multicast port where to listen for queries
+    public static final int DEFAULT_JOLOKIA_MULTICAST_PORT = 24884;
 
     // Utility class
     private MulticastUtil() {
     }
 
+    private static <T> T getOrDefault(ConfigKey configKey, String sysEnvKey, T defaultValue) {
+        String property = System.getProperty("jolokia." + configKey.getKeyValue());
```
Comment (rhuss) [inline agent/core/src/main/java/org/jolokia/discovery/MulticastUtil.java]: This lookup is already performed upwards in the stack and shouldn't happen here.

File: agent/core/src/main/java/org/jolokia/discovery/MulticastUtil.java [hunk 4]
```diff
@@ -17,22 +18,40 @@
...
+               return (T) Integer.valueOf(property);
+            } else {
+                return (T) property;
+            }
+        }
+        return defaultValue;
+    }
+
     static MulticastSocket newMulticastSocket(InetAddress pAddress, LogHandler pLogHandler) throws IOException {
+        String multicastGroup = getOrDefault(ConfigKey.MULTICAST_GROUP, "JOLOKIA_MULTICAST_GROUP", DEFAULT_JOLOKIA_MULTICAST_GROUP);
```
Comment (rhuss) [inline agent/core/src/main/java/org/jolokia/discovery/MulticastUtil.java]: This utility class is not the proper place for looking up configuration values. Instead this method should have two addition parameters, the group and the port. And feed in by the caller of this method.

Comment (rhuss) [reply agent/core/src/main/java/org/jolokia/discovery/MulticastUtil.java]: So the [method initializing the multicast thread](https://github.com/rhuss/jolokia/blob/ec1cebda618e68d12c9b1906820564408cf950e1/agent/core/src/main/java/org/jolokia/http/AgentServlet.java#L198) should pass through the `Configuration` object to the `MulticastSocketListenerThread` where it then be used to call this utility method.

I know its a bit tedious, but it helps in keeping the configuration handling separate.

**Comment by rhuss** [comment]
btw, if you do a rebase to the latest Jolokia master, the CI run will be fixed.

**Comment by graben** [comment]
I'm working on it but it will make the change more difficult and big.

**Comment by graben** [comment]
@rhuss : Ready to merge? :-)

**Comment by rhuss** [comment]
sorry, just back from a conference. will give it a review asap.

**Pull request merged**